v2/tools/generator/internal/codegen/pipeline/resource_registration_file.go

```diff
@@ -378,6 +391,75 @@ func createGetKnownStorageTypesFunc(
 	return f.DefineFunc()
 }
 
+func (r *ResourceRegistrationFile) createGetResourceExtensions(context *astmodel.CodeGenerationContext) (dst.Decl, error) {
+
+	return createGetResourceExtensionsImpl(
+		context,
+		r.resourceExtensions,
+		"getResourceExtensions",
+		"returns a list of Resource Extension types")
+
+}
+
+func createGetResourceExtensionsImpl(codeGenerationContext *astmodel.CodeGenerationContext, resources []astmodel.TypeName, funcName string, funcComment string) (dst.Decl, error) {
+	// Sort the resources for a deterministic file layout
+	sort.Slice(resources, orderByImportedTypeName(codeGenerationContext, resources))
+	packageRef, err := codeGenerationContext.GetImportedPackageName(astmodel.GenRuntimeReference)
+	if err != nil {
+		return nil, err
+	}
+	resultIdent := dst.NewIdent("result")
+	resultVar := astbuilder.LocalVariableDeclaration(
+		resultIdent.String(),
+		&dst.ArrayType{
+			Elt: &dst.SelectorExpr{
```

Comment:
So instead of building this `SelectorExpr` yourself, it's actually cleaner to define a `TypeName` in `astmodel` in the `std_references.go` file. Like this: `SecretReferenceType             = MakeTypeName(GenRuntimeReference, "SecretReference")`

Then you can just use the type by doing `astmodel.SecretReferenceType.AsType(codeGenerationContext)` rather than having to import the pkgRef and build the selector yourself.

Reply:
I think this will save you from having to do this above too:
```
	packageRef, err := codeGenerationContext.GetImportedPackageName(astmodel.GenRuntimeReference)
	if err != nil {
		return nil, err
	}
```